File: android/app/src/main/java/com/reviewanything/app/data/model/Note.kt
package com.reviewanything.app.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.Relation

@Entity(tableName = "notes")
data class Note(
    @PrimaryKey(autoGenerate = true)
    val id: Int = 0,
    val filePath: String = "",
    val fileName: String = "",
    val section: String = "",
    val content: String = "",
    val contentHash: String = "",
    val createdAt: Long = System.currentTimeMillis()
)

File: windows/ReviewAnything/Models/Note.cs
namespace ReviewAnything.Models;

public class Note
{
    public int Id { get; set; }
    public string FilePath { get; set; } = "";
    public string FileName { get; set; } = "";
    public string Section { get; set; } = "";
    public string Content { get; set; } = "";
    public string ContentHash { get; set; } = "";
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public List<Chunk> Chunks { get; set; } = new();
}
